tests/Unit/Parallel/Test_PupStlCpp11.cpp

```diff
@@ -99,6 +106,24 @@ SPECTRE_TEST_CASE("Unit.Serialization.PupStlCpp11", "[Serialization][Unit]") {
     reader | serialization_target;
     CHECK(serialization_target == to_serialize);
   }
+  {
+    INFO("map with custom comparator");
+    std::map<std::pair<int, double>, double, PairComparator> map_to_serialize;
+    map_to_serialize.insert({std::make_pair(1, 2.0), 3.0});
+    map_to_serialize.insert({std::make_pair(3, 1.0), 1.5});
+    map_to_serialize.insert({std::make_pair(2, 6.0), 10.2});
```

Comment:
I'm not sure what this commit is about (maybe give a brief description somewhere?), but this test is definitely not doing what you intend.  The statement on this line doesn't do anything because it is a "duplicate" of the first key that was inserted.

Reply:
Oh, yeah, I see the mistake; I guess I shouldn't have gotten cute with the comparison operator -- I'll fix the comparator.
The commit is just adding a pup function that a) doesn't interfere with existing pup functions for STL containers -- because we'll want to use the charm built-in one once it works, and b) doesn't cause a compiler error for a std::map with a custom comparator. There is currently a charm bug (actually just fixed, but not yet in a version) that causes `reserve` to be called when the map has a custom comparator, and therefore doesn't compile, I'll add that information to the commit message as well

Reply:
OK, I see.  I think you could add an overload of `PUP::reserve_if_applicable`, but I like the separate function better.

(It was a valid comparator, just probably not one you'd want for a general map.  I can even assign it almost-reasonable meaning as the timelike (i.e., non-null) comparator for events in null coordinates in SR with one axis reversed. :))